pilot/pkg/proxy/envoy/v2/mesh.go

```diff
@@ -0,0 +1,990 @@
+				case DestinationRuleIP:
+					labelsToMatch =
+						newMapWithLabelValue(subset.GetLabels(), DestinationIP.AttrName(), labelValue)
+				case DestinationRuleName:
+					labelsToMatch =
+						newMapWithLabelValue(subset.GetLabels(), DestinationName.AttrName(), labelValue)
+				case DestinationRuleCIDR | DestinationRuleWildcard:
+					// Direct matches will not work. Scoping is done later.
+					labelsToMatch = subset.GetLabels()
+				}
+				newEpSubset := m.reverseAttrMap.getEndpointsMatching(labelsToMatch)
+				if ruleType == DestinationRuleCIDR || ruleType == DestinationRuleWildcard {
+					// For rule names that do not map to values that can be label matched
+					// scope the returned set to the type of rule.
+					newEpSubset = newEpSubset.scopeToRule(ruleType, labelValue, cidrNet)
+				}
+				m.subsetEndpoints.addEndpointSet(scopedSSName, newEpSubset)
+				m.reverseEpSubsets.addSubsetForEndpoints(newEpSubset, scopedSSName)
+				m.subsetDefinitions[scopedSSName] = subset
+			}
+		}
+	}
+	return errs
+}
+
+// NewEndpoint is a boiler plate function intended for platform Controllers to create a new Endpoint.
+// This method ensures all the necessary data required for creating subsets are correctly setup. It
+// also performs sorting of arrays etc, to allow stable results for reflect.DeepEquals() for quick
+// comparisons. The address	is the network address of this endpoint that must be accessible from __THIS__
+// Pilot, i.e. one importing the Endpoint into its Mesh via Reconcile(). If a remote exposes a gateway,
+// the remote pilot exposes the gateway's address. The gateway itself may have more than one Endpoints
+// behind it that are not directly network accessible from this Pilot. Similarly the the network port
+// of this endpoint that must be accessible from __THIS__ pilot. socketProtocol should be set to
+// TCP or UPD. Labels are properties of the workload, for example: pod labels in Kubernetes.
+func NewEndpoint(address string, port uint32, socketProtocol SocketProtocol, labels []EndpointLabel) (*Endpoint, error) {
+	var errs error
+	ipAddr := net.ParseIP(address)
+	if ipAddr == nil {
+		errs = multierror.Append(errs, fmt.Errorf("invalid IP address '%s'", address))
+	}
+	hasUID := false
+	svcFQDNs := []string{}
+	svcDomains := []string{}
+	miscLabels := make(map[string]string, len(labels))
+	for _, label := range labels {
+		if prohibitedAttrSet[label.Name] {
+			errs = multierror.Append(fmt.Errorf(
+				"prohibited use of Istio destination label '%s' for endpoint label", label.Name))
+			continue
+		}
+		switch label.Name {
+		case DestinationService.AttrName():
+			svcFQDNs = append(svcFQDNs, label.Value)
+		case DestinationDomain.AttrName():
+			svcDomains = append(svcDomains, label.Value)
+		case DestinationUID.AttrName():
+			hasUID = true
+			fallthrough
+		default:
+			oldValue, found := miscLabels[label.Name]
+			if found {
+				errs = multierror.Append(fmt.Errorf(
```

Comment:
Here and in the rest of this function, you're missing the first `errs` argument:

```go
errs = multierror.Append(errs, fmt.Errorf(...))
```

Reply:
done